### PR TITLE
fix healthcheck path

### LINKED ISSUE #2273: healthcheck not working with custom --web.path
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?

Run traefik with cli options only and changed `--web.path` to the custom value `/traefik`. 

### What did you expect to see?

traefik healthcheck should work if `--web.path` is given as parameter, but it doesn't.
When i calling the `/traefik/ping` url with curl, it works. 

### What did you see instead?

Bad healthcheck status: 404 Not Found

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.0
Codename:     roquefort
Go version:   go1.9.1
Built:        2017-10-16_04:52:07PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```json
{
 "GraceTimeOut": 10000000000,
 "Debug": false,
 "CheckNewVersion": true,
 "AccessLogsFile": "",
 "AccessLog": null,
 "TraefikLogsFile": "",
 "LogLevel": "ERROR",
 "EntryPoints": {},
 "Cluster": null,
 "Constraints": [],
 "ACME": null,
 "DefaultEntryPoints": [],
 "ProvidersThrottleDuration": 2000000000,
 "MaxIdleConnsPerHost": 200,
 "IdleTimeout": 0,
 "InsecureSkipVerify": false,
 "RootCAs": null,
 "Retry": null,
 "HealthCheck": {
  "Interval": 30000000000
 },
 "RespondingTimeouts": null,
 "ForwardingTimeouts": null,
 "Docker": null,
 "File": null,
 "Web": null,
 "Marathon": null,
 "Consul": null,
 "ConsulCatalog": null,
 "Etcd": null,
 "Zookeeper": null,
 "Boltdb": null,
 "Kubernetes": null,
 "Mesos": null,
 "Eureka": null,
 "ECS": null,
 "Rancher": null,
 "DynamoDB": null,
 "ConfigFile": ""
}
```
```docker-compose
version: "3"
services:
  traefik:
    image: traefik:1.4.0-alpine
    command:
      # - --debug
      - --loglevel=INFO
      - --checknewversion=true
      - --insecureskipverify=true
      - --web
      - --web.path=/traefik
      - --web.address=:8080
      - --web.readonly
      - --web.statistics
      - --defaultentrypoints=http,https
      - "--entrypoints=Name:http Address::80 ProxyProtocol:true Compress:true Redirect.EntryPoint:https"
      - "--entryPoints=Name:https Address::443 TLS Compress:true ProxyProtocol:true"
      - --docker
      - --docker.watch
      - --docker.exposedbydefault=false
      - --docker.swarmmode
    ports:
      - 80:80
      - 443:443
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - traefik:/data
    deploy:
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 2
      update_config:
        delay: 5s
        failure_action: rollback
        monitor: 10s
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=org_traefik"
        - "traefik.port=8080"
        - "traefik.protocol=http"
        - "traefik.frontend.rule=PathPrefix:/traefik"
volumes:
  traefik:
networks:
  traefik:


### If applicable, please paste the log output in debug mode (`--debug` switch)

```
(paste your output here)
```

### PR DESCRIPTION
**Pull request by emilevauge**
### Description
This PR fixes healthcheck path.

Fixes #2273

### CODE REVIEW

File: configuration/configuration.go [hunk 1]
```diff
@@ -80,6 +80,58 @@ type GlobalConfiguration struct {
...
+	if gc.Debug {
+		gc.LogLevel = "DEBUG"
+	}
+	if gc.Web != nil {
+		if gc.Web.Path == "" {
+			gc.Web.Path = "/"
+		}
+		if gc.Web.Path[len(gc.Web.Path)-1:] != "/" {
+			gc.Web.Path += "/"
+		}
```
Comment (nmengin) [inline configuration/configuration.go]: Can not use a single _if_ block ? 

```go
               if gc.Web.Path == "" || gc.Web.Path[len(gc.Web.Path)-1:] != "/"  {
 			gc.Web.Path += "/"
 		}
```

Comment (emilevauge) [reply configuration/configuration.go]: Fixed :)

File: configuration/configuration.go [hunk 2]
```diff
@@ -80,6 +80,55 @@ type GlobalConfiguration struct {
...
+
+		if gc.Rancher.Metadata != nil && len(gc.Rancher.Metadata.Prefix) == 0 {
+			gc.Rancher.Metadata.Prefix = "latest"
+		}
+	}
+	if gc.Debug {
+		gc.LogLevel = "DEBUG"
+	}
+	if gc.Web != nil {
+		if gc.Web.Path == "" || gc.Web.Path[len(gc.Web.Path)-1:] != "/"  {
```
Comment (ldez) [inline configuration/configuration.go]: `gc.Web.Path[len(gc.Web.Path)-1:] != "/"` can be replace by a `!strings.HasSuffix(gc.Web.Path, "/")`

Comment (ldez) [reply configuration/configuration.go]: the 2 if can be merged:
```go
if gc.Web != nil && (gc.Web.Path == "" || !strings.HasSuffix(gc.Web.Path, "/")) {
	gc.Web.Path += "/"
}
```

**Pull request merged**